encoding/basics-and-workflow.md

```diff
@@ -2,36 +2,584 @@
+
+psy-rd sets the Psy-RDO and Psy-Trellis.
+For an explanation on how it works, read [this doom9 post](psy-rd_post).
+For anime you usually go with `--psy-rd 0.72:0`,
+and tweak the Psy-RDO settings (first value),
+while generally leaving the Psy-Trellis settings (second value) alone.
+Higher values can help with better grain retention.
+
+For newer versions of x264,
+it is important to set the `--output-depth 10` flag
+when encoding in 10bit colordepth.
+Otherwise it will encode in 8bit.
+Likewise, when encoding in a 4:4:4 colorspace,
+make sure you set `--output-csp i444`.
+
+With those things in mind,
+this is what a common command looks like:
 
 ```sh
-$ vspipe.exe script.vpy -y - | x264.exe --demuxer y4m --some example --parameters here --output video.264 -
+$ vspipe FTOP12.vpy --y4m - | x264 --demuxer y4m -o FTOP12.264 - --preset veryslow --crf 14 --keyint 240 --min-keyint 23 --ref 16 --bframes 16  --aq-mode 3 --aq-strength 0.70 --qcomp 0.70 --no-dct-decimate --no-fast-pskip --psy-rd 0.72:0.0 --output-depth 10
 ```
 
-Editors for VapourSynth usually have inbuilt support for encoding
-scripts you wrote. Use `%encode --y4m <clip_variable>` in Yuuno or the GUI
-provided by VSEdit.
+Editors for VapourSynth usually have inbuilt support
+for encoding scripts you wrote.
+Use `%encode --y4m <clip_variable>` in Yuuno
+or the GUI provided by VSEdit.
 
 
 ## Transcoding Audio
 
-examples for qaac, flac
+Before getting into transcoding audio,
+it is important to learn about
+the difference between
+"lossy" and "lossless" compression.
+
+If you're transcoding in a lossy format,
+you are chucking away data
+deemed unnecessary by the encoder.
+This is the kind of encoding you do with video,
+however it can also be done with audio.
+The benefit of this is that
+you'll end up with smaller files,
+however this might cause a reduction in quality.
+
+Lossless compression is a form of compression where,
+upon decoding, none of the data is lost.
+This is fantastic for preservation.
+However, the results can also turn up
+pretty big, very quickly.
+
+Picking between lossy and lossless encoding
+comes down to what you prefer.
+Be it having smaller filesizes
+with possibly next to no perceivable difference from the original,
+or bigger files with as much information as possible.
+
+You should never encode from a lossy audio codec
+to a lossless one, though.
+This will only increase the filesize for zero gain.
+Likewise, you want to avoid lossy-to-lossy encoding
+as much as you possibly can.
+The more data you chuck away,
+the worse it will sound,
+because of [generational loss][generational_loss].
+
+There are generally three audio codecs
+that you will use:
 
+* FLAC
+* AAC
+* opus
+
+FLAC is a **F**ree **L**ossless **A**udio **C**odec.
+This is the go-to for lossless audio encoding,
+as it's better at compression
+than other lossless codecs (like PCM).
+You can encode to fLAC using ffmpeg
+with the following command:
+
+`$ ffmpeg -i input.wav output.flac`
```

Comment:
These will probably look better in proper blocks. You could also use the `{% term %}` blocks from a plugin, as shown in `CONTIBUTING.md`.